Inconsistency when using .to_numpy()

Linux 0.8.0.dev561

```
X = dt.fread("weather_missing.csv")[-15:, 'EvapMM']
A = X.to_pandas().values
B = X.to_numpy()
print(A)
print(B)
assert np.equal(A, B).all()
```

```
[[6.6]
 [7. ]
 [9.6]
 [3.8]
 [6.8]
 [8.4]
 [5.8]
 [nan]
 [nan]
 [6.8]
 [2.4]
 [nan]
 [9.8]
 [9.2]
 [8.2]]
[[6.6]
 [7.0]
 [9.6]
 [3.8]
 [6.8]
 [8.4]
 [5.8]
 [nan]
 [nan]
 [6.8]
 [2.4]
 [--]
 [9.8]
 [9.2]
 [8.2]]
```
```
AssertionError: assert False
 +  where False = <bound method MaskedArray.all of masked_array(\n  data=[[True],\n        [True],\n        [True],\n        [True],\n       ...     [False],\n        [False],\n        [ True],\n        [False],\n        [False],\n        [False]],\n  fill_value=True)>()
 +    where <bound method MaskedArray.all of masked_array(\n  data=[[True],\n        [True],\n        [True],\n        [True],\n       ...     [False],\n        [False],\n        [ True],\n        [False],\n        [False],\n        [False]],\n  fill_value=True)> = masked_array(\n  data=[[True],\n        [True],\n        [True],\n        [True],\n        [True],\n        [True],\n        ...      [False],\n        [False],\n        [ True],\n        [False],\n        [False],\n        [False]],\n  fill_value=True).all
 +      where masked_array(\n  data=[[True],\n        [True],\n        [True],\n        [True],\n        [True],\n        [True],\n        ...      [False],\n        [False],\n        [ True],\n        [False],\n        [False],\n        [False]],\n  fill_value=True) = <ufunc 'equal'>(array([[6.6],\n       [7. ],\n       [9.6],\n       [3.8],\n       [6.8],\n       [8.4],\n       [5.8],\n       [nan],\n       [nan],\n       [6.8],\n       [2.4],\n       [nan],\n       [9.8],\n       [9.2],\n       [8.2]]), masked_array(\n  data=[[6.6],\n        [7.0],\n        [9.6],\n        [3.8],\n        [6.8],\n        [8.4],\n        [5.8],...     [False],\n        [False],\n        [ True],\n        [False],\n        [False],\n        [False]],\n  fill_value=1e+20))
 +        where <ufunc 'equal'> = np.equal
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -85,6 +85,8 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
   datatable `bool8` column when pandas DataFrame is converted into a datatable
   Frame (#1730).
 
+- Fixed conversion to numpy of a view Frame which contains NAs (#1738).
+
 
 ### Changed
 
@@ -120,7 +122,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 - Thanks to everyone who helped make `datatable` more stable by discovering
   and reporting bugs that were fixed in this release:
 
-  [arno candel][] (#1619, #1730),
+  [arno candel][] (#1619, #1730, #1738),
   [antorsae][] (#1639),
   [pasha stetsenko][] (#1672, #1694, #1695, #1697, #1703, #1705)
 

--- a/c/column_fw.cc
+++ b/c/column_fw.cc
@@ -317,9 +317,10 @@ RowIndex FwColumn<T>::join(const Column* keycol) const {
 template <typename T>
 void FwColumn<T>::fill_na_mask(int8_t* outmask, size_t row0, size_t row1) {
   const T* tdata = elements_r();
-  for (size_t i = row0; i < row1; ++i) {
-    outmask[i] = ISNA<T>(tdata[i]);
-  }
+  ri.iterate(row0, row1, 1,
+    [&](size_t i, size_t j) {
+      outmask[i] = (j == RowIndex::NA) || ISNA<T>(tdata[j]);
+    });
 }
 
 

--- a/tests/test_dt.py
+++ b/tests/test_dt.py
@@ -875,6 +875,14 @@ def test_tonumpy_with_NAs_random(seed, numpy):
     assert ar.T.tolist() == data
 
 
+@pytest.mark.usefixtures("numpy")
+def test_tonumpy_with_NAs_view():
+    # See issue #1738
+    X = dt.Frame(A=[5.7, 2.3, None, 4.4, 9.8, None])[1:, :]
+    a = X.to_numpy()
+    assert a.tolist() == [[2.3], [None], [4.4], [9.8], [None]]
+
+
 
 #-------------------------------------------------------------------------------
 # [0, 0] (conversion to scalar python variable)

